base/multidimensional.jl

```diff
@@ -617,32 +617,33 @@ end
 # contiguous multidimensional indexing: if the first dimension is a range,
 # we can get some performance from using copy_chunks!
 
-@inline function setindex!(B::BitArray, X::BitArray, I0::UnitRange{Int})
+@inline function setindex!(B::BitArray, X::BitArray, I0::Union{Colon,UnitRange{Int}})
     @boundscheck checkbounds(B, I0)
-    l0 = length(I0)
+    l0 = index_lengths(B, I0)[1]
     setindex_shape_check(X, l0)
     l0 == 0 && return B
     f0 = first(I0)
     copy_chunks!(B.chunks, f0, X.chunks, 1, l0)
     return B
 end
 
-@inline function setindex!(B::BitArray, x, I0::UnitRange{Int})
+@inline function setindex!(B::BitArray, x, I0::Union{Colon,UnitRange{Int}})
     @boundscheck checkbounds(B, I0)
     y = Bool(x)
-    l0 = length(I0)
+    l0 = index_lengths(B, I0)[1]
     l0 == 0 && return B
     f0 = first(I0)
     fill_chunks!(B.chunks, y, f0, l0)
     return B
 end
 
-@inline function setindex!(B::BitArray, X::BitArray, I0::UnitRange{Int}, I::Union{Int,UnitRange{Int}}...)
+@inline function setindex!(B::BitArray, X::BitArray, I0::UnitRange{Int}, I0::Union{Colon,UnitRange{Int}}, I::Union{Int,UnitRange{Int},Colon}...)
```

Comment:
Will the repeated `I0` here cause a bootstrap failure?  If so - this is a friendly reminder to squash. :)